src/Engine/Manager/ResourceManager.cpp

```diff
@@ -74,6 +74,21 @@ TextureAsset* ResourceManager::CreateTextureAsset(const std::string& name) {
     return textureAssets[name].textureAsset;
 }
 
+TextureAsset* ResourceManager::CreateTextureAsset(const std::string& name, Video::Texture2D* texture) {
+    if (textureAssets.find(name) == textureAssets.end()) {
+        TextureAsset* textureAsset = new TextureAsset();
+        textureAsset->Load(name, texture);
+        textureAssets[name].textureAsset = textureAsset;
+        textureAssetsInverse[textureAsset] = name;
+        textureAssets[name].count = 1;
+    }
+    else {
+        textureAssets[name].count++;
+    }
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetTextureAssetInstanceCount' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)